shared/schema.js

```diff
@@ -0,0 +1,64 @@
+/*
```

Comment:
Design decision: I've tried very hard to client-side validate using JSON schemas, and it's surprisingly difficult to get useful validation data out of ajv (despite what everyone says online). I have found [yup](https://github.com/jquense/yup) to be very useful, and have developed snippets to enable validation at a Feathers level. That said, JSON schemas are probably a lot more flexible. Wdyt?

(In any case, I think it'd be good to move the validation to a [hook](https://feathers-plus.github.io/v1/feathers-hooks-common/#validateschema)--this centralizes the business logic to hooks, so it's easier to find things imo.)